app/src/main/java/org/listenbrainz/android/ui/screens/profile/createdforyou/CreatedForYouScreen.kt

```diff
@@ -0,0 +1,242 @@
+package org.listenbrainz.android.ui.screens.profile.createdforyou
+
+import android.content.Context
+import android.content.Intent
+import androidx.compose.animation.AnimatedContent
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.launch
+import org.listenbrainz.android.model.SocialUiState
+import org.listenbrainz.android.model.createdForYou.CreatedForYouPlaylist
+import org.listenbrainz.android.model.playlist.PlaylistTrack
+import org.listenbrainz.android.ui.components.ErrorBar
+import org.listenbrainz.android.ui.components.SuccessBar
+import org.listenbrainz.android.ui.screens.feed.SocialDropdownDefault
+import org.listenbrainz.android.ui.screens.profile.ProfileUiState
+import org.listenbrainz.android.ui.theme.ListenBrainzTheme
+import org.listenbrainz.android.util.Utils.getCoverArtUrl
+import org.listenbrainz.android.viewmodel.SocialViewModel
+import org.listenbrainz.android.viewmodel.UserViewModel
+
+@Composable
+fun CreatedForYouScreen(
+    snackbarState: SnackbarHostState,
+    socialViewModel: SocialViewModel,
+    userViewModel: UserViewModel,
+    goToArtistPage: (String) -> Unit,
+) {
+    val uiState by userViewModel.uiState.collectAsState()
+    val context = LocalContext.current
+    val scope = rememberCoroutineScope()
+    val socialUiState by socialViewModel.uiState.collectAsState()
+    CreatedForYouScreen(uiState = uiState,
```

Comment:
Good work with this abstraction, just add a preview and it would be chef's kiss.